astronomer/providers/core/sensors/external_task.py

```diff
@@ -115,20 +115,19 @@ class ExternalDeploymentTaskSensorAsync(HttpSensorAsync):
 
     def execute(self, context: Context) -> None:
         """Defers trigger class to poll for state of the job run until it reaches a failure state or success state"""
-        if not self.poke(context=context):
-            self.defer(
-                timeout=self.execution_timeout,
-                trigger=ExternalDeploymentTaskTrigger(
-                    http_conn_id=self.http_conn_id,
-                    method=self.hook.method,
-                    endpoint=self.endpoint,
-                    data=self.request_params,
-                    headers=self.headers,
-                    extra_options=self.extra_options,
-                    poke_interval=self.poke_interval,
-                ),
-                method_name="execute_complete",
-            )
+        self.defer(
```

Comment:
I think we decided to check before deferring because we were hitting some issues with Kubernetes executer when a task was spending very small time on triggerer.  @Lee-W / @phanikumv what do you think about this change?

Reply:
agree

Reply:
I tried it with KubernetesExecutor too, looks working just fine to me.

Dockerfile:

```
FROM quay.io/astronomer/astro-runtime:8.8.0
USER root
RUN pip install git+https://github.com/abhishekbhakat/astronomer-providers.git@fix-externaldeploymenttasksensorasync
USER astro
```

Should I be looking for anything specific ?